### ISSUE TITLE
Regression: Routes in ActionView::TestCase for 5.0.0.beta1

### ISSUE THREAD

**Issue by jcoyne** [rails/rails#22815]
Testing a view that calls `url_for(params.merge(only_path: false))`

This test worked in Rails 4.2, but fails in 5.0.0.beta1
```
     ActionView::Template::Error:

       arguments passed to url_for can't be handled. Please require routes or provide your own implementation

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/helpers/url_helper.rb:38:in `url_for'

     # ./app/views/catalog/index.atom.builder:12:in `block in __home_travis_build_projectblacklight_blacklight_app_views_catalog_index_atom_builder___1142194302259704669_90130360'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/builder-3.2.2/lib/builder/xmlbase.rb:175:in `call'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/builder-3.2.2/lib/builder/xmlbase.rb:175:in `_nested_structures'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/builder-3.2.2/lib/builder/xmlbase.rb:68:in `tag!'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/builder-3.2.2/lib/builder/xmlbase.rb:93:in `method_missing'

     # ./app/views/catalog/index.atom.builder:5:in `__home_travis_build_projectblacklight_blacklight_app_views_catalog_index_atom_builder___1142194302259704669_90130360'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/template.rb:159:in `block in render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/activesupport-5.0.0.beta1/lib/active_support/notifications.rb:166:in `instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/template.rb:353:in `instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/template.rb:157:in `render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/template_renderer.rb:54:in `block (2 levels) in render_template'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/activesupport-5.0.0.beta1/lib/active_support/notifications.rb:164:in `block in instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/activesupport-5.0.0.beta1/lib/active_support/notifications/instrumenter.rb:21:in `instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/activesupport-5.0.0.beta1/lib/active_support/notifications.rb:164:in `instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/template_renderer.rb:53:in `block in render_template'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/template_renderer.rb:61:in `render_with_layout'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/template_renderer.rb:52:in `render_template'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/template_renderer.rb:14:in `render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/renderer.rb:42:in `render_template'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/renderer/renderer.rb:23:in `render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/helpers/rendering_helper.rb:32:in `render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/test_case.rb:199:in `render'

     # /home/travis/.rvm/gems/ruby-2.2.4/gems/actionview-5.0.0.beta1/lib/action_view/test_case.rb:120:in `render'
```

**Comment by jcoyne** [comment]
In 4.2.5, the code path goes through https://github.com/rails/rails/blob/master/actionview/lib/action_view/routing_url_for.rb#L85-L90, but it doesn't get there in 5.0.0

**Comment by jcoyne** [comment]
This was due to using the released gem rails-controller-testing (version 0.0.3) rather than the master branch.

**Comment by WojtekKruszewski** [comment]
I believe that's https://github.com/rails/rails-controller-testing/issues/12
